testy/stateManager.py

```diff
@@ -0,0 +1,42 @@
+# import multiprocessing
+import sys
+
+from runtimeUtil import *
+
+# TODO:
+# 0. modify self.state to acually store (name, value) pairs
+class StateManager(object):
+
+  """input is a multiprocessing.Queue object to support multiple
+  processes requesting state data
+  """
+  def __init__(self, badThingsQueue, inputQueue, runtimePipe):
+    self.initRobotState()
+    self.badThingsQueue = badThingsQueue
+    self.input = inputQueue
+    # map process names to pipes
+    self.processMapping = {PROCESS_NAMES.RUNTIME: runtimePipe}
+
+  def initRobotState(self):
+    self.state = [5]
+
+  def addPipe(self, processName, pipe):
+    self.processMapping[processName] = pipe
+    pipe.send(SM_COMMANDS.READY)
+
+  def start(self):
+    while True:
+      request = self.input.get(block=True)
```

Comment:
Should protect against malicious requests, by checking that request is actually well-formed (i.e. a tuple/list)